packages/realm_dart/lib/src/app.dart

```diff
@@ -219,16 +219,17 @@ class App implements Finalizable {
 
   /// Returns the current value of the base URL used to communicate with the server.
   @experimental
-  Uri? get baseUrl {
-    return Uri.tryParse(realmCore.getBaseUrl(this) ?? '');
+  Uri get baseUrl {
+    return Uri.parse(realmCore.getBaseUrl(this));
   }
 
   /// Temporarily overrides the [baseUrl] value from [AppConfiguration] with a new [baseUrl] value
-  /// used for communicating with the server.
+  /// used for communicating with the server. If set to `null`, the app will revert to the default
+  /// base url.
```

Comment:
I'm wondering if the DX and the intent would be clearer if the SDKs provided a separate zero-arguments API for reverting to the default base URL, rather than passing `null` (or empty string for the SDKs accepting a string) to the update-API, even if that's how the Core API currently works. Thoughts?

Reply:
I think providing zero-argument API will be more confusing as it's not obvious what `app.updateBaseUrl()` will update it to. In this case I feel `null` == `default` is a reasonable assumption developers can make. Alternatively, we could provide another method `app.resetBaseUrl()` to the same effect, but I'm not convinced this functionality justifies 3 methods on the app class.

I'm leaning toward leaving the API proposed in this PR (i.e. accepting null/"" for reseting the url to the default) and then re-evaluating based on user feedback/broader discussion as we revisit the API and move it out of the experimental phase.

Reply:
With zero-argument API I meant a different (3rd) API taking zero args, e.g. `resetBaseUrl()` like you mentioned. The flexible part atm is that it's experimental, so alrighty we can see at a later point if it'd make sense or not 🙂

Reply:
I prefer to just have a single `updateBaseUrl` method and have a `null` argument reset it.

One thing to consider is to align on the API across SDKs. From the rumblings in the office it seems we are at risk of diverting a bit.

Reply:
From the original design we should have accepted only strings (or Uri), so for TS, we're currently only allowing the empty string as a reset (passing `null` would do the same but we haven't added it to the type). If we should also allow `null` to align more, we could do that as well. For the JS SDK, in the future it may be more idiomatic for us to either use a 3rd API or an options arg like `{ reset: true }`.

Reply:
I'd say we should allow `null` here - I don't think it's obvious that you can pass an empty string to achieve the reset behavior, while `null`/`undefined` is special enough to convey that better.